addons/website_sale_loyalty/models/sale_order.py

```diff
@@ -156,7 +156,17 @@ def get_promo_code_success_message(self, delete=True):
             request.session.pop('successful_code')
         return code
 
+    def _is_to_delete_discount(self, line, set_qty):
+        self.ensure_one()
+        return set_qty == 0 and line.coupon_id and line.reward_id and line.reward_id.reward_type == 'discount'
+
     def _cart_update(self, *args, **kwargs):
+        product_id, set_qty = kwargs['product_id'], kwargs.get('set_qty', None)
```

Comment:
```suggestion
        product_id, set_qty = kwargs['product_id'], kwargs.get('set_qty')
```
You can use the default here, it's `None` =]